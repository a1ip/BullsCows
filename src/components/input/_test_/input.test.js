import React from "react"
import renderer from 'react-test-renderer'

import { InputGuess } from "../Input"

describe('Input component', () => {

    let testValue = '0'
    const handleChange = (e) => {
        testValue = e.target.value
    }
    const InputTest = renderer
        .create(<InputGuess order={0} onChange={handleChange} value={5}/>)

    it('renders correctly', () => {
        expect(InputTest.toJSON()).toMatchSnapshot()
    })

    it('changes properly', () => {
        const component = InputTest
        let changeComponent = component
        expect(changeComponent.toJSON()).toMatchSnapshot()
        expect(testValue).toBe('0')
        InputTest.root.props.onChange({
            target: { value: '6' }
        });
        // re-rendering
        changeComponent = component
        expect(testValue).toBe('6')
        expect(changeComponent.toJSON()).toMatchSnapshot()

    })
})