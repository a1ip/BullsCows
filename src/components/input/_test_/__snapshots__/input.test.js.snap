// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component changes properly 1`] = `
<input
  autocomplete="off"
  className="input-guess"
  maxLength={1}
  name={0}
  onChange={[Function]}
  value={5}
/>
`;

exports[`Input component changes properly 2`] = `
<input
  autocomplete="off"
  className="input-guess"
  maxLength={1}
  name={0}
  onChange={[Function]}
  value={5}
/>
`;

exports[`Input component renders correctly 1`] = `
<input
  autocomplete="off"
  className="input-guess"
  maxLength={1}
  name={0}
  onChange={[Function]}
  value={5}
/>
`;
